java/core/src/main/java/co/worklytics/psoxy/rules/msft/PrebuiltSanitizerRules.java

```diff
@@ -275,6 +275,112 @@ public class PrebuiltSanitizerRules {
             REDACT_CALENDAR_ODATA_LINKS);
 
 
+    static final String MS_TEAMS_PATH_REGEX_TEAMS = "^/(v1.0|beta)/teams";
+    static final String MS_TEAMS_PATH_REGEX_TEAMS_ALL_CHANNELS = "^/(v1.0|beta)/teams/[^/]*/allChannels";
+    static final String MS_TEAMS_PATH_REGEX_USERS_CHATS = "^/(v1.0|beta)/users/[^/]*/chats";
+    static final String MS_TEAMS_PATH_REGEX_TEAMS_CHANNELS_MESSAGES = "^/(v1.0|beta)/teams/[^/]*/channels/[^/]*/messages";
+    static final String MS_TEAMS_PATH_REGEX_TEAMS_CHANNELS_MESSAGES_DELTA = "^/(v1.0|beta)/teams/[^/]*/channels/[^/]*/messages/delta";
+    static final String MS_TEAMS_PATH_REGEX_USERS_CHATS_MESSAGES = "^/(v1.0|beta)/users/chats/[^/]*/messages";
+    static final String MS_TEAMS_PATH_REGEX_COMMUNICATIONS_CALLS = "^/(v1.0|beta)/communications/calls/[^/]*";
+    static final String MS_TEAMS_PATH_REGEX_COMMUNICATIONS_CALLRECORDS = "^/(v1.0|beta)/communications/callRecords/[^/]*";
+    static final String MS_TEAMS_PATH_REGEX_USERS_ONLINEMEETINGS = "^/(v1.0|beta)/users/[^/]*/onlineMeetings";
+
+
+    static final List<Endpoint>  MS_TEAMS_ENDPOINTS = Arrays.asList(
+        Endpoint.builder()
+            .pathRegex(MS_TEAMS_PATH_REGEX_TEAMS)
+            .transform(Transform.Redact.builder()
+                .jsonPath("$..displayName")
+                .jsonPath("$..description")
+                .build())
+            .allowedQueryParams(List.of("$select","$top","$skipToken","$filter", "$count"))
+            .build(),
+        Endpoint.builder()
+            .pathRegex(MS_TEAMS_PATH_REGEX_TEAMS_ALL_CHANNELS)
+            .allowedQueryParams(List.of("$select","$filter"))
+            .transform(Transform.Redact.builder()
+                .jsonPath("$..displayName")
+                .jsonPath("$..description")
+                .build())
+            .build(),
+        Endpoint.builder()
+            .pathRegex(MS_TEAMS_PATH_REGEX_USERS_CHATS)
+            .allowedQueryParams(List.of("$select","$top","$skipToken", "$filter",  "$orderBy", "$expand"))
+            .build(),
+        Endpoint.builder()
+            .pathRegex(MS_TEAMS_PATH_REGEX_TEAMS_CHANNELS_MESSAGES)
+            .allowedQueryParams(List.of("$select","$top","$skipToken","$expand"))
+            .transform(Transform.Redact.builder()
+                .jsonPath("$..from.user.displayName")
+                .jsonPath("$..body.content")
+                .jsonPath("$..attachments")
+                .jsonPath("$..mentions[*].mentionText")
+                .jsonPath("$..mentions[*].mentioned.user.displayName")
+                .jsonPath("$..eventDetail.teamDescription")
+                .jsonPath("$..eventDetail.initiator.user.displayName")
+                .build())
+            .build(),
```

Comment:
I've found anything what need to pseudonymize. Not email or etc, just pure name and I just redact this fields.

Reply:
I've spoken @eschultink  about this and we should pseudonymize user.id. @jlorper mentioned we probably should pseudonymize all, but because other MS endpoint not pseudonymize everything - we probably cannot do this for MS Teams. 

@aperez-worklytics ^

Reply:
Done.